### PR TITLE
feat!: used Emotion for Anchor

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- [x] Related to designs: https://app.abstract.com/projects/a4a24600-c8c9-11e8-a177-81a5fa367641/branches/328ef9fe-09cf-43f4-8dd1-40947f5d4fb4/commits/3a51a13f42daf13efbfd15dc96f34ad00a72052d/files/FA9AF2C4-35B2-4183-8E93-A9D84A525613/layers/7598FF01-E0C7-4FAD-A505-B6136FA4DC91?mode=design _(throwback!)_
- [x] Related to JIRA ticket: WEB-1177
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

Migrates the old clunky `Anchor` component to a shockingly simple CSS-in-JS solution. The API is no longer clunky so it can be marked as `stable` in its story.

Note: it still relies on global styles on the page for its color...

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@5.7.3-alpha.a2cb85.0<br/>@codecademy/gamut@17.4.1-alpha.a2cb85.0<br/>@codecademy/styleguide@18.10.2-alpha.a2cb85.0<br/></pre>

File: packages/styleguide/stories/Core/Typography/Anchor.stories.mdx [hunk 1]
```diff
@@ -35,5 +33,3 @@ It applies `text-decoration: underline` per our accessibility guidelines to dist
 </Canvas>
 
 <Props story="Anchor" />
-
-We know the API is a little unwieldy; we're thinking it'll get improved when we switch to CSS-in-JS...
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Core/Typography/Anchor.stories.mdx:39]: we did the thing! 🎊

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fbe4ff1f0c9f26084320692--gamut-preview.netlify.app>https://5fbe4ff1f0c9f26084320692--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fbe4ff1f0c9f26084320692>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.3-alpha.087d1c.0<br/>@codecademy/gamut@17.4.5-alpha.087d1c.0<br/>@codecademy/styleguide@19.0.3-alpha.087d1c.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@6.0.3-alpha.1fabd0.0<br/>@codecademy/gamut@17.4.5-alpha.1fabd0.0<br/>@codecademy/styleguide@19.0.3-alpha.1fabd0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc5759d0da6792fa93d08b5--gamut-preview.netlify.app>https://5fc5759d0da6792fa93d08b5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc5759d0da6792fa93d08b5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5fc5760c8400cb2975eb43e3--gamut-preview.netlify.app>https://5fc5760c8400cb2975eb43e3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5fc5760c8400cb2975eb43e3>Deploy Logs</a></p>

**Review by codecaaron — APPROVED** [review]
<3

**Pull request merged**